PWGHF/D2H/TableProducer/dataCreatorDV0Reduced.cxx

```diff
@@ -0,0 +1,184 @@
+  using TracksPIDWithSelAndMc = soa::Join<TracksPIDWithSel, aod::McTrackLabels>;
+  using CandsDFiltered = soa::Filtered<soa::Join<aod::HfCand3Prong, aod::HfSelDplusToPiKPi>>;
+
+  Filter filterSelectCandidates = (aod::hf_sel_candidate_dplus::isSelDplusToPiKPi >= selectionFlagD);
+
+  Preslice<CandsDFiltered> candsDPerCollision = aod::track_association::collisionId;
+  Preslice<aod::TrackAssoc> trackIndicesPerCollision = aod::track_association::collisionId;
+  Preslice<aod::V0Datas> candsV0PerCollision = aod::track_association::collisionId;
+
+  HistogramRegistry registry{"registry"};
+
+  void init(InitContext const&)
+  {
+    // histograms
+    constexpr int kNBinsEvents = kNEvent;
+    std::string labels[kNBinsEvents];
+    labels[Event::Processed] = "processed";
+    labels[Event::NoDV0Selected] = "without DV0 pairs";
+    labels[Event::DV0Selected] = "with DV0 pairs";
+    static const AxisSpec axisEvents = {kNBinsEvents, 0.5, kNBinsEvents + 0.5, ""};
+    registry.add("hEvents", "Events;;entries", HistType::kTH1F, {axisEvents});
+    for (int iBin = 0; iBin < kNBinsEvents; iBin++) {
+      registry.get<TH1>(HIST("hEvents"))->GetXaxis()->SetBinLabel(iBin + 1, labels[iBin].data());
+    }
+    registry.add("hMassDToPiKPi", "D^{#minus} candidates;inv. mass (p^{#minus} K^{#plus} #pi^{#minus}) (GeV/#it{c}^{2});entries", {HistType::kTH1F, {{100, 1.7, 2}}});
+    registry.add("hMassV0", "K0^{s} candidates;inv. mass (#pi^{#plus}#pi^{#minus}) (GeV/#it{c}^{2});entries", {HistType::kTH1F, {{100, 0.35, 0.65}}});
+    registry.add("hPtD", "D^{#minus} candidates;D^{#minus} candidate #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}});
+    registry.add("hPtV0", "V0 candidates;V0 candidate #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}});
+    registry.add("hMassDs1", "Ds1 candidates;Ds1 candidate #inv. mass (GeV/#it{c}^{2});entries", {HistType::kTH1F, {{100, 2., 3.}}});
+  }
+
+  void runDataCreation(aod::Collision const& collision,
+                       CandsDFiltered const& candsD,
+                       aod::V0Datas const& V0s)
+  {
+    // helpers for ReducedTables filling
+    int indexHfReducedCollision = lfReducedCollision.lastIndex() + 1;
+    // std::map where the key is the V0.globalIndex() and
+    // the value is the V0 index in the table of the selected v0s
+    std::map<int64_t, int64_t> selectedV0s;
+    bool fillHfReducedCollision = false;
+
+    for (const auto& candD : candsD) {
+      bool fillHfCandD = false;
+      float invMassD = hfHelper.invMassDplusToPiKPi(candD);
+      std::array<float, 3> pVecD = candD.pVector();
+
+      for (const auto& v0 : V0s) {
+        std::array<float, 3> pVecV0 = {v0.pxpos() + v0.pxneg(), v0.pypos() + v0.pyneg(), v0.pzpos() + v0.pzneg()};
+        float ptV0 = sqrt(pVecV0[0] * pVecV0[0] + pVecV0[1] * pVecV0[1]);
+        auto invMass2DV0 = RecoDecay::m2(std::array{pVecD, pVecV0}, std::array{MassDPlus, MassK0Short});
```

Comment:
In order to compute the DV0 invariant mass we should propagate the V0 to the PV. You can see an example here:
https://github.com/AliceO2Group/O2Physics/blob/84fcfb8279ce172a7bd47e1d2e55e183fc7adb40/EventFiltering/PWGHF/HFFilter.cxx#L599-L603

Reply:
Thank you!